library/psa_crypto.c

```diff
@@ -1010,122 +557,76 @@ psa_status_t psa_copy_key_material_into_slot( psa_key_slot_t *slot,
     if( status != PSA_SUCCESS )
         return( status );
 
-    memcpy( slot->data.key.data, data, data_length );
+    memcpy( slot->key.data, data, data_length );
     return( PSA_SUCCESS );
 }
 
-/** Import key data into a slot.
- *
- * `slot->type` must have been set previously.
- * This function assumes that the slot does not contain any key material yet.
- * On failure, the slot content is unchanged.
- *
- * Persistent storage is not affected.
- *
- * \param[in,out] slot  The key slot to import data into.
- *                      Its `type` field must have previously been set to
- *                      the desired key type.
- *                      It must not contain any key material yet.
- * \param[in] data      Buffer containing the key material to parse and import.
- * \param data_length   Size of \p data in bytes.
- *
- * \retval #PSA_SUCCESS
- * \retval #PSA_ERROR_INVALID_ARGUMENT
- * \retval #PSA_ERROR_NOT_SUPPORTED
- * \retval #PSA_ERROR_INSUFFICIENT_MEMORY
- */
-static psa_status_t psa_import_key_into_slot( psa_key_slot_t *slot,
-                                              const uint8_t *data,
-                                              size_t data_length )
+psa_status_t psa_import_key_into_slot(
```

Comment:
It's the equivalent of the "_internal()" functions for the import operation. I kept the original name because "_into_slot" compared to "_internal" brings some useful information to me. The function is called from the driver wrapper to dispatch depending on the type of the key. I agree that eventually the dispatch should be in the driver wrapper but that's additional changes, more commits and it is already quite big.

Otherwise, the documentation is not gone, it is in psa_crypto_core.h.

Reply:
Existence: ok then, makes sense for it to be future work.

Documentation: ah, right, makes perfect sense.